src/deps_hack/src/lib.rs

```diff
@@ -137,18 +139,47 @@ pub fn default_storage() -> k8s_openapi::apimachinery::pkg::api::resource::Quant
     k8s_openapi::apimachinery::pkg::api::resource::Quantity("10Gi".to_string())
 }
 
-pub fn default_storage_class_name() -> Option<String> {
-    None
-}
-
 #[derive(Debug, Clone, serde::Deserialize, serde::Serialize, schemars::JsonSchema)]
 pub struct RabbitmqClusterPersistenceSpec {
-    #[serde(rename = "storageClassName", default = "default_storage_class_name")]
+    #[serde(rename = "storageClassName", default)]
     pub storage_class_name: Option<String>,
     #[serde(default = "default_storage")]
     pub storage: k8s_openapi::apimachinery::pkg::api::resource::Quantity,
 }
 
+/// Provides the ability to override the generated manifest of several child resources.
+#[derive(Default, Debug, Clone, serde::Deserialize, serde::Serialize, schemars::JsonSchema)]
+pub struct RabbitmqClusterOverrideSpec {
+	/// Override configuration for the RabbitMQ StatefulSet.
+    #[serde(rename = "statefulSet")]
+	pub stateful_set: Option<RabbitmqOverrideStatefulSet>,
+}
+
+/// Allows for the manifest of the created StatefulSet to be overwritten with custom configuration.
+#[derive(Debug, Clone, serde::Deserialize, serde::Serialize, schemars::JsonSchema)]
+pub struct RabbitmqOverrideStatefulSet {
+	pub spec: Option<RabbitmqOverrideStatefulSetSpec>,
+}
+
+/// StatefulSetSpec contains a subset of the fields included in k8s.io/api/apps/v1.StatefulSetSpec.
+/// Every field is made optional.
+#[derive(Debug, Clone, serde::Deserialize, serde::Serialize, schemars::JsonSchema)]
+pub struct RabbitmqOverrideStatefulSetSpec {
```

Comment:
I don't think we have to follow the same hierarchy as the official rabbitmq controller. We can promote `pod_management_policy` and `persistent_volume_claim_retention_policy` to the top level without having these middle layers.

Reply:
What’s the benefit?

Reply:
keep the interface simple